src/open_producten/producttypen/serializers/onderwerp.py

```diff
@@ -0,0 +1,106 @@
+from django.core.exceptions import ValidationError
+from django.db import transaction
+
+from rest_framework import serializers
+
+from open_producten.producttypen.models import (
+    Onderwerp,
+    ProductType,
+    UniformeProductNaam,
+)
+from open_producten.utils.serializers import build_array_duplicates_error_message
+
+from .vraag import VraagSerializer
+
+
+class SimpleProductTypeSerializer(serializers.ModelSerializer):
+    uniforme_product_naam = serializers.SlugRelatedField(
+        slug_field="uri", queryset=UniformeProductNaam.objects.all()
+    )
+
+    class Meta:
+        model = ProductType
+        exclude = (
+            "onderwerpen",
+            # "organisaties",
+            # "locaties",
+            # "contacten",
+        )
+
+
+class OnderwerpSerializer(serializers.ModelSerializer):
+    hoofd_onderwerp = serializers.PrimaryKeyRelatedField(
+        queryset=Onderwerp.objects.all(),
+        allow_null=True,
+    )
+    product_typen = SimpleProductTypeSerializer(many=True, read_only=True)
+    vragen = VraagSerializer(many=True, read_only=True)
+
+    product_type_ids = serializers.PrimaryKeyRelatedField(
+        many=True,
+        queryset=ProductType.objects.all(),
+        default=[],
+        write_only=True,
+        source="product_typen",
+    )
+
+    class Meta:
+        model = Onderwerp
+        exclude = ("path", "depth", "numchild")
+
+    def _handle_relations(self, instance, product_typen: list[ProductType]):
+        errors = dict()
+        if product_typen is not None:
+            build_array_duplicates_error_message(
+                product_typen, "product_type_ids", errors
+            )
+            instance.product_typen.set(product_typen)
+
+        if errors:
+            raise serializers.ValidationError(errors)
```

Comment:
```suggestion
            errors =build_array_duplicates_error_message(product_typen)
            
            if errors:
                raise serializers.ValidationError(  {"product_type_ids": errors})
            
            instance.product_typen.set(product_typen)
```


`if errors` should be moved before the ` instance.product_typen.set(product_typen)`

You could also remove the field and errors arguments from `build_array_duplicates_error_message` just return the list of errors. Not sure if this is slower.

Reply:
Also if product_typen is none, errors will always be none, so there is no need to test